src/conversion.jl

```diff
@@ -37,7 +54,7 @@ Find the conversion factor from unit `t` to unit `s`, e.g., `convfact(m, cm) ==
             "exponents and/or SI prefixes in units"
         ))
     end
-    return :($result)
+    return :($(result isa AbstractFloat ? UnitConversionFactor(result) : result))
```

Comment:
```suggestion
    return result isa AbstractFloat ? UnitConversionFactor(result) : result
```
I tested it and it worked to return just a value instead of an expression.

Reply:
@sostock, can I commit this?

Reply:
Yes